TELCODOCS-2015: Adding tables that break down the network flow by platform (#85385)
openshift/openshift-docs: modules/network-flow-matrix.adoc

[id="network-flow-matrix_{context}"]
= {product-title} network flow matrix

The following network flow matrixes describe the ingress flows to {product-title} services for the following environments:

* {product-title} on bare metal 

> * {sno} on bare metal
> * {product-title} on {aws-first}
> * {sno} on {aws-short}

Reviewer: `{sno-caps}`?
```suggestion
* {sno-caps} on bare metal
* {product-title} on {aws-first}
* {sno-caps} on {aws-short}
```

Use the information in the appropriate network flow matrix to help you manage ingress traffic for your specific environment. You can restrict ingress traffic to essential flows to improve network security.

Additionally, consider the following dynamic port ranges when managing ingress traffic for both bare metal and cloud environments:

* `9000-9999`: Host level services
* `30000-32767`: Kubernetes node ports
* `49152-65535`: Dynamic or private ports

To view or download the complete raw CSV content for an environment, see the following resources:

* link:https://raw.githubusercontent.com/openshift-kni/commatrix/release-4.18/docs/stable/raw/bm.csv[{product-title} on bare metal]

> * link:https://raw.githubusercontent.com/openshift-kni/commatrix/release-4.18/docs/stable/raw/bm-sno.csv[{sno} on bare metal]
> 
> * link:https://raw.githubusercontent.com/openshift-kni/commatrix/release-4.18/docs/stable/raw/aws.csv[{product-title} on {aws-short}]
> 
> * link:https://raw.githubusercontent.com/openshift-kni/commatrix/release-4.18/docs/stable/raw/aws-sno.csv[{sno} on {aws-short}]

Reviewer: ```suggestion
* link:https://raw.githubusercontent.com/openshift-kni/commatrix/release-4.18/docs/stable/raw/bm-sno.csv[{sno-caps} on bare metal]

* link:https://raw.githubusercontent.com/openshift-kni/commatrix/release-4.18/docs/stable/raw/aws.csv[{product-title} on {aws-short}]

* link:https://raw.githubusercontent.com/openshift-kni/commatrix/release-4.18/docs/stable/raw/aws-sno.csv[{sno-caps} on {aws-short}]
```

[NOTE]
====

> The network flow matrixes describe ingress traffic flows for a base {product-title} or {sno} installation. It does not describe network flows for additional components, such as optional Operators available from the Red Hat Marketplace. The matrixes do not apply for Hosted-Control-Plane, MicroShift, or standalone clusters.

Reviewer 2: ```suggestion
The network flow matrixes describe ingress traffic flows for a base {product-title} or {sno} installation. It does not describe network flows for additional components, such as optional Operators available from the Red Hat Marketplace. The matrixes do not apply for {hcp}, MicroShift, or standalone clusters.
```

Hosted control planes is written as -- 
- **h**osted control planes: In subsequent occurrences ;  attribute {hcp}
- **H**osted control planes: Beginning of sentence, title, and heading ; {hcp-capital}

====

[id="network-flow-matrix-control_{context}"]
.Control plane node base flows
[%header,format=csv]
|===
include::https://raw.githubusercontent.com/openshift-kni/commatrix/release-4.18/docs/stable/unique/common-master.csv[]
|===

[id="network-flow-matrix-worker_{context}"]
.Worker node base flows
[%header,format=csv]
|===
include::https://raw.githubusercontent.com/openshift-kni/commatrix/release-4.18/docs/stable/unique/common-worker.csv[]
|===

[id="network-flow-matrix-bm_{context}"]
== Additional network flows for {product-title} on bare metal

In addition to the base network flows, the following matrix describes the ingress flows to {product-title} services that are specific to {product-title} on bare metal.

.{product-title} on bare metal
[%header,format=csv]
|===
include::https://raw.githubusercontent.com/openshift-kni/commatrix/release-4.18/docs/stable/unique/bm.csv[]
|===

[id="network-flow-matrix-sno_{context}"]
== Additional network flows for {sno} on bare metal

In addition to the base network flows, the following matrix describes the ingress flows to {product-title} services that are specific to {sno} on bare metal.

> .{sno} on bare metal 

Reviewer: ```suggestion
.{sno-caps} on bare metal 
```

[%header,format=csv]
|===
include::https://raw.githubusercontent.com/openshift-kni/commatrix/release-4.18/docs/stable/unique/bm-sno.csv[]
|===

[id="network-flow-matrix-aws_{context}"]
== Additional network flows for {product-title} on {aws-short}

In addition to the base network flows, the following matrix describes the ingress flows to {product-title} services that are specific to {product-title} on {aws-short}.

.{product-title} on AWS 
[%header,format=csv]
|===
include::https://raw.githubusercontent.com/openshift-kni/commatrix/release-4.18/docs/stable/unique/aws.csv[]
|===

[id="network-flow-matrix-aws-sno_{context}"]
== Additional network flows for {sno} on {aws-short}

In addition to the base network flows, the following matrix describes the ingress flows to {product-title} services that are specific to {sno} on {aws-short}.

> .{sno} on AWS 

Reviewer: ```suggestion
.{sno-caps} on AWS 
```

//link:https://raw.githubusercontent.com/openshift/openshift-docs/main/snippets/network-flow-matrix.csv[this resource].

* link:https://raw.githubusercontent.com/rohennes/openshift-docs/TELCODOCS-2015/snippets/communication-matrix-bm.csv[{product-title} on bare metal]

* link:https://raw.githubusercontent.com/rohennes/openshift-docs/TELCODOCS-2015/snippets/communication-matrix-bm-sno.csv[{sno} on bare metal]

* link:https://raw.githubusercontent.com/rohennes/openshift-docs/TELCODOCS-2015/snippets/communication-matrix-aws.csv[{product-title} on {aws-short}]

* link:https://raw.githubusercontent.com/rohennes/openshift-docs/TELCODOCS-2015/snippets/communication-matrix-aws-sno.csv[{sno} on {aws-short}]

[NOTE]
====
The network flow matrixes describe ingress traffic flows for a base {product-title} or {sno} installation. It does not describe network flows for additional components, such as optional Operators available from the Red Hat Marketplace. The matrixes do not apply for Hosted-Control-Plane, MicroShift, or standalone clusters.
====

[id="network-flow-matrix-common_{context}"]
== Network flow matrix base flows

> The following matrixes describe the ingress flows to {product-title} services that are common to the following environments:

Reviewer 3: I'm not sure about mentioning the list of the platforms, I was thinking this should talk about standard OCP and mention here for SNO reference control plane list only.
Should we also mention that additional flows per platform can be found next?

Author: I removed the list of platforms.

I added a note about SNO.

I don't know if we need to say additional flows are later in the content, I've updated the headings so I think it should be apparent.

* {product-title} on bare metal 
* {sno} on bare metal
* {product-title} on {aws-first}
* {sno} on {aws-short}

[id="network-flow-matrix-control_{context}"]
.Control plane node base flows
[%header,format=csv]
|===

> include::https://raw.githubusercontent.com/openshift-kni/commatrix/main/docs/4.18/commatrix-common-master.csv[]

Reviewer 3: I think we should point to a specific branch and not main, release-4.18.
In addition, I'm updating 4.18 to stable to fit with OCP standard https://github.com/openshift-kni/commatrix/pull/73

Author: Done

|===

[id="network-flow-matrix-worker_{context}"]
.Worker node base flows
[%header,format=csv]
|===
include::https://raw.githubusercontent.com/openshift-kni/commatrix/main/docs/4.18/commatrix-common-worker.csv[]
|===

[id="network-flow-matrix-bm_{context}"]

> == Network flow matrix for {product-title} on bare metal

Reviewer 3: Should this header mention this are additional ports?
